### PR TITLE
Fixed Scoping issues (mainly with Fortran)

### PR DESCRIPTION
**Pull request by brtnfld**
Went down the scoping rabbit hole. Fixed all the missing scope enumerations in the Fortran tests, which lead to getting rid of the fixed formatting, Reworked CONCATENATE to work with gfortran (see cgnstypes_f03.h.in), CGNS-249

Not Fun, thankfully we have _find_ and _Perl._

### CODE REVIEW

**Comment by MicK7** [comment]
I am not really familiar with the scoping feature so I trust others to review all the holes.
Thanks for the not so fun dig up

**Comment by MicK7** [comment]
@gsjaardema @brtnfld  Do you think this will require a minor release 4.2.1 ?

**Comment by brtnfld** [comment]
I would not think a minor release is needed since PR#265 fixed a test, and this fixes a previous bug/tests already in 4.1.2 (and 4.2.0); the CGNS library itself is not affected.

**Comment by brtnfld** [comment]
@gsjaardema, are you going to have time to review this, or should I just merge it?

File: src/Test_UserGuideCode/Fortran_code/write_con2zn_str.F90 [hunk 1]
```diff
@@ -1,6 +1,7 @@
       program write_con2zn_str
       use cgns
       implicit none
+#include "cgnstypes_f03.h"
```
Comment (gsjaardema) [inline src/Test_UserGuideCode/Fortran_code/write_con2zn_str.F90:4]: Doesn't look ilke this one is needed.

**Review by gsjaardema — APPROVED** [review]
Looks good.  Tried to skim every file and everything looked like a reasonable change.  

The `cgns_io.c`file that was changed for the `mpi_initialized` looks out-of-place in these changes.  But it looks like a needed change.

**Comment by brtnfld** [comment]
I don't know why the mpi_initialized change is showing up, that change is already in develop.

**Pull request merged**